avocado/core/test.py

```diff
@@ -166,9 +167,9 @@ def str_filesystem(self):
         elif len(self.str_uid) <= len(test_id_fs):   # full uid
             return astring.string_to_safe_path(self.str_uid + self.str_variant)
         else:       # not even uid could be stored in fs
-            raise AssertionError('Test uid is too long to be stored on the '
-                                 'filesystem: "%s"\nFull Test ID: "%s"'
-                                 % (self.str_uid, str(self)))
+            raise ValueError('Test ID is too long to be stored on the '
+                             'filesystem: "%s"\nFull Test ID: "%s"'
+                             % (self.str_uid, str(self)))
```

Comment:
Yes, ValueError is better here, but even better IMO would be `RuntimeError`, because this is not strictly bound to a single value, it's more related to the combination of operating system and uid, what do you think?

Reply:
OK, `RuntimeError`  is indeed a better fit here.